src/Rings/MPolyQuo.jl

```diff
@@ -1885,36 +1885,50 @@ symbols(A::MPolyQuoRing) = symbols(base_ring(A))
 # Tensor products of rings
 ########################################################################
 
-function tensor_product(A::MPolyRing, B::MPolyRing)
+function tensor_product(A::MPolyRing, B::MPolyRing; use_product_ordering::Bool = false)
   kk = coefficient_ring(A)
   @assert kk === coefficient_ring(B) "coefficient rings do not coincide"
   res, a, b = polynomial_ring(kk, symbols(A), symbols(B); cached=false)
+  if use_product_ordering
+    o = induce(gens(res)[1:ngens(A)], default_ordering(A))*induce(gens(res)[ngens(A) + 1:end], default_ordering(B))
+    set_default_ordering!(res, o)
+  end
   return res, hom(A, res, a), hom(B, res, b)
 end
 
-function tensor_product(A::MPolyRing, B::MPolyQuoRing)
+function tensor_product(A::MPolyRing, B::MPolyQuoRing; use_product_ordering::Bool = false)
   R = base_ring(B)
   AR, inc_A, inc_R = tensor_product(A, R)
   I = ideal(AR, inc_R.(gens(modulus(B))))
-  res, pr = quo(R, I)
-  return res, compose(inc_A, pr), hom(B, res, pr.(inc_R.(gens(R))))
+  res, pr = quo(AR, I)
+  if use_product_ordering
+    o = induce(gens(AR)[1:ngens(A)], default_ordering(A))*induce(gens(AR)[ngens(A) + 1:end], default_ordering(B))
+    set_default_ordering!(AR, o)
+  end
+  return res, hom(A, res, pr.(inc_A.(gens(A))); check = false), hom(B, res, pr.(inc_R.(gens(R))); check = false)
 end
 
-function tensor_product(A::MPolyQuoRing, B::MPolyQuoRing)
+function tensor_product(A::MPolyQuoRing, B::MPolyQuoRing; use_product_ordering::Bool = false)
   RA = base_ring(A)
   RB = base_ring(B)
   P, inc_A, inc_B = tensor_product(RA, RB)
   I = ideal(P, inc_A.(gens(modulus(A)))) + ideal(P, inc_B.(gens(modulus(B))))
-  res, pr = quo(R, I)
+  res, pr = quo(P, I)
+  if use_product_ordering
+    o = induce(gens(P)[1:ngens(A)], default_ordering(A))*induce(gens(P)[ngens(A) + 1:end], default_ordering(B))
+    set_default_ordering!(P, o)
+  end
   return res, hom(A, res, pr.(inc_A.(gens(RA))); check=false), hom(B, res, pr.(inc_B.(gens(RB))); check=false)
 end
 
-function tensor_product(A::MPolyQuoRing, B::MPolyRing)
+function tensor_product(A::MPolyQuoRing, B::MPolyRing; use_product_ordering::Bool = false)
   RA = base_ring(A)
   P, inc_A, inc_B = tensor_product(RA, B)
   I = ideal(P, inc_A.(gens(modulus(A))))
-  res, pr = quo(R, I)
-  return res, hom(A, res, pr.(inc_A.(gens(RA))); check=false), compose(inc_B, pr)
```

Comment:
@HechtiDerLachs I replaced the `compose` by a "manual" construction of the map because `pr` is a `MapFromFunc` so the result of `compose` is a generic `CompositeMap` with which one can't do much.

Reply:
Looks fine. If it works, it works.